include/nbl/video/IPhysicalDevice.h

```diff
@@ -267,6 +267,20 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             uint8_t storageBufferViewAtomic : 1u; // imageBuffer
             uint8_t accelerationStructureVertex : 1u;
 
+            SFormatBufferUsage()
+                : isInitialized(0)
+            {}
+
+            SFormatBufferUsage(core::bitflag<asset::IBuffer::E_USAGE_FLAGS> usages)
+                : isInitialized(1),
+                vertexAttribute(usages.hasFlags(asset::IBuffer::EUF_VERTEX_BUFFER_BIT)),
+                bufferView(usages.hasFlags(asset::IBuffer::EUF_UNIFORM_TEXEL_BUFFER_BIT)),
+                storageBufferView(usages.hasFlags(asset::IBuffer::EUF_STORAGE_TEXEL_BUFFER_BIT)),
+                // Other way of getting these flags?
+                accelerationStructureVertex(usages.hasFlags(asset::IBuffer::EUF_ACCELERATION_STRUCTURE_BUILD_INPUT_READ_ONLY_BIT)),
+                storageBufferViewAtomic(usages.hasFlags(asset::IBuffer::EUF_STORAGE_TEXEL_BUFFER_BIT))
```

Comment:
ok, this is gonna backfire, because every time you'll declare a `EUF_STORAGE_TEXEL_BUFFER_BIT` usage, it will think you're gonna do atomics, and then promote everything to R32_UINT or R32_INT

you need a `const bool useForAtomic` extra parameter for the ctor, and do an && with the `storageBufferView`